pkg/csi/cinder/identityserver.go

```diff
@@ -47,11 +47,15 @@ func (ids *identityServer) GetPluginInfo(ctx context.Context, req *csi.GetPlugin
 }
 
 func (ids *identityServer) Probe(ctx context.Context, req *csi.ProbeRequest) (*csi.ProbeResponse, error) {
-
-	_, err := openstack.CreateOpenStackProvider()
+	klog.V(5).Infof("Probe() called with req %+v", req)
+	oProvider, err := openstack.GetOpenStackProvider()
 	if err != nil {
-		klog.V(3).Infof("Failed to CreateOpenStackProvider: %v", err)
-		return nil, status.Error(codes.FailedPrecondition, "Failed to communicate with openstack")
+		klog.V(3).Infof("Failed to GetOpenStackProvider: %v", err)
+		return nil, status.Error(codes.FailedPrecondition, "Failed to retrieve openstack provider")
+	}
+	if _, err := oProvider.CheckBlockStorageAPI(); err != nil {
```

Comment:
sorry, from bug description I got this
multiple times, we can optimize this call to a smaller check, maybe calling a minimal Cinder API
but seem this only add a call instead optimize? maybe I missed something..thanks

Reply:
Hi @ramineni @jichenjc 
IIUC, this change reduces number of call for `CreateOpenStackProvider()` because `CreateOpenStackProvider()` is called only one time when kubelet or control manager start up, right?
But I suppose there is a case that Keystone authentication token in OpenStack Client will be expired if we don't renew `NewOpenStackClient()` after passing certain period.

Reply:
@mtanino right, CreateOpenStackProvider() is called at startup , so every probe call doesnt need to create it again.
>> But I suppose there is a case that Keystone authentication token in OpenStack Client will >> be expired if we don't renew NewOpenStackClient() after passing certain period.
I believe automatic renewal of session should be possible, but Ill look into to it again. Thanks for bringing it up.

Reply:
@jichenjc hope your question is answered now.

Reply:
ok, thanks for the info @ramineni

Reply:
@ramineni @jichenjc 
> I believe automatic renewal of session should be possible, but Ill look into to it again. 
> Thanks for bringing it up.
Thank you for the comment. For the purpose of automatic renewal of session, I think we can use following PR to configure probe period of keystone.
Could you also consider if this works for your improvement of Probe() func?

https://github.com/kubernetes/cloud-provider-openstack/pull/947

Reply:
@mtanino it should automatically renew token, https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/cloudprovider/providers/openstack/openstack.go#L331

Reply:
Thanks for the info. This param works for me.
```
	// Persistent service, so we need to be able to renew tokens.
	ao.AllowReauth = true
```